File: src/LCP77.W2ZX4X.1000565/solution.ts
function runeReserve (runes: number[]): number {
    runes.sort((a, b) => a - b);
    let result = 1;
    let index = 1;
    let count = 1;
    while (index < runes.length) {
        if (runes[index] - runes[index - 1] > 1) {
            count = 1;
        } else {
            count++;
            result = Math.max(result, count);
        }
        index++;
    }
    return result;
}

File: src/LCP77.W2ZX4X.1000565/solution.java
class Solution {
    public int runeReserve(int[] runes) {
        Arrays.sort(runes);
        int result = 0;
        int index = 0;
        while (index<runes.length){
            int start = index;
            int last = runes[index++];
            while (index<runes.length && runes[index]-last<2){
                last = runes[index++];
            }
            result = Math.max(result,index-start);
        }
        return result;
    }
}